hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JobRepository.java

```diff
@@ -270,17 +278,23 @@ long newExecutionId(long jobId) {
      * @throws JobNotFoundException if the JobRecord is not found
      * @throws IllegalStateException if the JobResult is already present
      */
-    void completeJob(long jobId, JobMetrics terminalMetrics, String coordinator, long completionTime, Throwable error) {
+    void completeJob(
+        long jobId, JobMetrics terminalMetrics, String coordinator, long completionTime, Throwable error
+    ) {
         JobRecord jobRecord = getJobRecord(jobId);
         if (jobRecord == null) {
             throw new JobNotFoundException(jobId);
         }
 
         JobConfig config = jobRecord.getConfig();
         long creationTime = jobRecord.getCreationTime();
-        JobResult jobResult = new JobResult(jobId, terminalMetrics, config, coordinator, creationTime, completionTime,
+        JobResult jobResult = new JobResult(jobId, config, coordinator, creationTime, completionTime,
                 error != null ? error.toString() : null);
 
+        JobMetrics prevMetrics = jobMetrics.putIfAbsent(jobId, terminalMetrics);
```

Comment:
We should just use `put`, not `putIfAbsent`: if a value was there, that was probably some error and we at least correct it. Even the logging 2 lines below says so.